include/deal.II/base/point.h

```diff
@@ -549,8 +549,9 @@ inline DEAL_II_CUDA_HOST_DEV
   Point<dim, Number>::operator/(const OtherNumber factor) const
 {
   Point<dim, typename ProductType<Number, OtherNumber>::type> tmp;
+  const Number inverse_factor = Number(1.) / factor;
```

Comment:
Here and in the place below, I tried to think through whether we want to use `Number` or `OtherNumber` as type. I came to the conclusion that what we really want (I think!) is
```
  const decltype(this->operator[](0)/factor)  inverse_factor = ...;
```